core/dbt/parser/parsing-vs-compilation-vs-runtime.md

```diff
@@ -0,0 +1,151 @@
+# Parsing vs. Compilation vs. Runtime
+
+## Context: Why this doc?
+
+There’s a lot of confusion about what dbt does at parse time vs. compile time / runtime. Even that separation is a relative simplification: parsing includes multiple steps, and while there are some distinctions between "compiling" and "running" a model, the two are **very** closely related.
+
+It's come up many times before, and we expect it will keep coming up! A decent number of bug reports in `dbt-core` are actually rooted in a misunderstanding of when configs are resolved, especially when folks are using pre/post hooks, or configs that alter materialization behavior (`partitions`, `merge_exclude_columns`, etc).
+
+So, here goes.
+
+## What is "parsing"?
+
+**In a sentence:** dbt reads all the files in your project, and constructs an internal representation of the project ("manifest").
+
+To keep it really simple, let’s say this happens in two steps: "Parsing" and "Resolving."
+
+### Parsing
+
+As a user, you write models as SQL + YAML. dbt wants to understand that model as a Python object, defined by an internal data structure. It also wants to know its dependencies and configuration (= its place in the DAG). dbt reads your code **for that one model,** and attempts to construct that object, raising a **validation** error if it can’t.
```

Comment:
```suggestion
As a user, you write models as SQL + YAML. dbt wants to understand each model as a Python object, defined by an internal data structure. It also wants to know its dependencies and configuration (= its place in the DAG). dbt reads your code **for that one model,** and attempts to construct that object, raising a **validation** error if it can’t.
```